festim/temperature/temperature_solver.py

```diff
@@ -75,9 +75,18 @@ def create_functions(self, materials, mesh, dt=None):
         self.v_T = f.TestFunction(V)
 
         if self.transient:
```

Comment:
I think that if `transient` is `True` and `initial_condition` is `None` , then the `else` on line 86 will be triggered. 

Maybe here we need to check that `self.initial_condition` is not `None` :

```suggestion
        if self.transient and self.initial_condition:
```